pkg/registry/core/service/storage/rest_test.go

```diff
@@ -621,6 +621,55 @@ func TestServiceRegistryUpdate(t *testing.T) {
 	}
 }
 
+func TestServiceRegistryUpdateUnspecifiedAllocations(t *testing.T) {
+	ctx := genericapirequest.NewDefaultContext()
+	storage, server := NewTestREST(t, []api.IPFamily{api.IPv4Protocol})
+	defer server.Terminate(t)
+
+	// Need a clusterIP and NodePort allocated.
+	svc := svctest.MakeService("foo", svctest.SetTypeLoadBalancer, svctest.SetExternalTrafficPolicy(api.ServiceExternalTrafficPolicyTypeLocal))
+	obj, err := storage.Create(ctx, svc.DeepCopy(), rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Expected no error: %v", err)
+	}
+	createdSvc := obj.(*api.Service)
+	if createdSvc.Spec.ClusterIP == "" {
+		t.Fatalf("expected ClusterIP to be set")
+	}
+	if len(createdSvc.Spec.ClusterIPs) == 0 {
+		t.Fatalf("expected ClusterIPs to be set")
+	}
+	if createdSvc.Spec.Ports[0].NodePort == 0 {
+		t.Fatalf("expected NodePort to be set")
+	}
+	if createdSvc.Spec.HealthCheckNodePort == 0 {
+		t.Fatalf("expected HealthCheckNodePort to be set")
+	}
+
+	// Update the selector - trivial.
```

Comment:
should we worry about mutating services types or some of these less common services nuances?

Reply:
If you mutate ServiceType and it no longer needs these values, then we don't want to carry them forward, right?  I thought about adding more here, but it feels redundant - we have tests around all sorts of updates already.

Reply:
sounds good, just being paranoid